### PR TITLE
feat: implement allowed denoms

### PR DESCRIPTION
**Pull request by johnletey**
Per the request of the Initia team, integrators of the `x/forwarding` module now have fine-tuned control over which denoms are automatically forwarded. Note that clearing a forwarding account to a fallback address will transfer all of the balance, regardless of allowed denoms.

## Summary by CodeRabbit

- **New Features**
	- Introduced the ability to fine-tune control over which denominations are forwarded.
	- Added new RPC methods for querying and setting allowed denominations.
	- Provided enhanced authority management functionalities.

- **Documentation**
	- Updated API documentation to reflect new features and functionalities.

- **Bug Fixes**
	- Implemented error handling for invalid authority and denominations input.

- **Refactor**
	- Enhanced internal logic for denomination and authority handling to improve performance and reliability.

- **Chores**
	- Updated scripts and configurations to support new authority account functionalities and denomination settings.

### CODE REVIEW

File: api/noble/forwarding/v1/tx.pulsar.go [hunk 1]
```diff
@@ -1942,6 +1942,906 @@ func (x *fastReflection_MsgClearAccountResponse) ProtoMethods() *protoiface.Meth
...
+}
+
+func (x *_MsgSetAllowedDenoms_2_list) NewElement() protoreflect.Value {
+	v := ""
+	return protoreflect.ValueOfString(v)
+}
+
+func (x *_MsgSetAllowedDenoms_2_list) IsValid() bool {
+	return x.list != nil
+}
```
[the remark this answers is not in the document]
Comment (johnletey) [reply api/noble/forwarding/v1/tx.pulsar.go:1989]: `tx.pulsar.go` is a generated file, and so is not relevant.

File: proto/noble/forwarding/v1/genesis.proto [hunk 2]
```diff
@@ -5,7 +5,8 @@ package noble.forwarding.v1;
 option go_package = "github.com/noble-assets/forwarding/v2/x/forwarding/types";
 
 message GenesisState {
-  map<string, uint64> num_of_accounts = 1;
-  map<string, uint64> num_of_forwards = 2;
-  map<string, string> total_forwarded = 3;
+  repeated string allowed_denoms = 1;
+  map<string, uint64> num_of_accounts = 2;
+  map<string, uint64> num_of_forwards = 3;
+  map<string, string> total_forwarded = 4;
```
[the remark this answers is not in the document]
Comment (johnletey) [reply proto/noble/forwarding/v1/genesis.proto:11]: No migration is needed, as this feature will be included in a major release of the module, and so we can have state breaking changes.

File: x/forwarding/types/genesis.go [hunk 3]
```diff
@@ -2,16 +2,24 @@ package types
...
-	return &GenesisState{}
+	return &GenesisState{
+		AllowedDenoms: []string{"*"},
+	}
 }
 
 func (gen *GenesisState) Validate() error {
+	if err := ValidateAllowedDenoms(gen.AllowedDenoms); err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (johnletey) [reply x/forwarding/types/genesis.go:21]: For now we only have E2E tests for the module, however if we do add unit tests later, we will definitely add this check!

File: x/forwarding/types/genesis.go [hunk 4]
```diff
@@ -36,3 +44,17 @@
...
+		return errors.New("wildcard can only be present by itself")
+	}
+
+	for _, denom := range denoms {
+		if strings.TrimSpace(denom) == "" {
+			return errors.New("cannot allow empty denom")
+		}
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (johnletey) [reply x/forwarding/types/genesis.go:62]: Implemented in https://github.com/noble-assets/forwarding/pull/13/commits/63da50d0a96aae777d0b42f9bcedeec1032fd688.

**Review by beer-1 — APPROVED** [review]
LGTM - thanks for the update!

**Pull request merged**